### DISCUSSION TITLE
Replace `bound` with `vm.assume` in invariant handlers

### DISCUSSION THREAD

**Discussion by smol-ninja** [sablier-labs/flow#243] [General]
`bound` and `seed` inputs in invariant handlers make it difficult to debug call sequences in a failed invariant. An example is shown below:

<img width="950" alt="Screenshot 2024-09-17 at 15 37 34" src="https://github.com/user-attachments/assets/deedf3d4-1943-48a7-8cf4-db4aa57eac97">

I also discovered that Foundry does not even return the correct sequence in failed test (the above sequence is not correct). The reason we use `bound` and `seed` inputs was to not waste any of the calls during invariant testing. However, it increases the cost to debug a failed invariant test.

-----------------

The screenshot below shows the sequence when I used `vm.assume` and fuzzed inputs in the handler functions.

<img width="945" alt="Screenshot 2024-09-17 at 15 37 20" src="https://github.com/user-attachments/assets/58487b9d-a380-4f3b-8c18-de035906d0dc">

You can see, the sequence and the function inputs are very clear and easy to follow. Thus, I propose to make the following changes:

- Replace `bound` with `vm.assume` in all handlers
- Use fuzzed inputs directly instead of constructing valid inputs using bound

PS: We can increase the numbers of runs and depth in CI to compensate for wasted calls.

cc @sablier-labs/solidity

**Comment by PaulRBerg** [comment]
Sounds good, feel free to open PRs directly for enhancements like this
